src/middlewares/piwik.js

```diff
@@ -0,0 +1,40 @@
+/* global Piwik */
+import {
+  PRE_UPLOAD_FILE
+} from '../actions'
```

Comment:
Not sure this middleware should depend on another file. It prevents to be easily extracted and reused elsewhere.